addons/web/controllers/home.py

```diff
@@ -83,9 +83,15 @@ def web_login(self, redirect=None, **kw):
         if request.httprequest.method == 'GET' and redirect and request.session.uid:
             return request.redirect(redirect)
 
-        # so it is correct if overloaded with auth="public"
-        if not request.uid:
-            request.update_env(user=odoo.SUPERUSER_ID)
+        # simulate hybrid auth=user/auth=public, despite using auth=none to be able
+        # to redirect users when no db is selected - cfr ensure_db()
+        if request.env.uid is None:
+            if request.session.uid is None:
+                # no user -> auth=public with specific website public user
+                request.env["ir.http"]._auth_method_public()
+            else:
+                # auth=user
+                request.update_env(user=request.session.uid)
```

Comment:
This else case cannot happen as the environment (`request.env`) is created using `request.session.uid` already.

https://github.com/odoo/odoo/blob/ae0a11a8ce3c35542ba5ae8f6366018b88ec40d5/odoo/http.py#L1546

Reply:
except that's it's reset by auth=none, it's the same logic as for the /web controller above

https://github.com/odoo/odoo/blob/ad8599c361e8f8916a0f37f22478c351a0a0ba9a/addons/web/controllers/home.py#L49-L50